kafka-agent/src/main/java/io/strimzi/kafka/agent/KafkaAgent.java

```diff
@@ -153,11 +162,19 @@ public void run() {
             boolean handleBrokerState() {
                 LOGGER.trace("Polling {}", brokerStateName);
                 boolean ready = false;
-                Integer running = Integer.valueOf(3);
+                Integer running = 3;
+                Integer unknown = 127;
                 Object value = brokerState.value();
 
-                if ((value instanceof Integer && running.equals(value))
-                        || (value instanceof Byte && running.equals(((Byte) value).intValue()))) {
+                boolean stateIsRunning = false;
+                if (value instanceof Integer) {
+                    stateIsRunning = inKRaftMode ? running.compareTo((Integer) value) <= 0 && !unknown.equals(value) : running.equals(value);
```

Comment:
For the broker readiness probe: `broker state >=3 && broker state != 127`, I think it applied to both ZK and KRaft mode brokers. @tombentley , could you confirm?

Reply:
I'm not following the comments here, @showuon and @tombentley are you saying that the behaviour I have here is correct, where for KRaft mode we mark a node with `broker state >=3 && broker state != 127` as ready, but with ZK-based broker we only mark ready if `broker state == 3`? Or are you saying I should update the ZK-based check to also check for >=3  && != 127.

Yeah I think those variable renames will make it clearer

Reply:
> I should update the ZK-based check to also check for >=3 && != 127.

That is what I'm saying.

Reply:
Ok, I've updated the PR to use the same check for both ZK-based and KRaft mode when checking the state and renamed the variables.